mailet/standard/src/test/java/org/apache/james/transport/mailets/UseHeaderRecipientsTest.java

```diff
@@ -0,0 +1,191 @@
+
+        String RCPT_1 = "abc1@apache1.org";
+        String RCPT_2 = "abc2@apache2.org";
+
+        FakeMail fakeMail = FakeMail.builder()
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.ANY_AT_JAMES2)
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .addToRecipient(RCPT_1, RCPT_2)
+                .build())
+            .build();
+
+        testee.service(fakeMail);
+
+        assertThat(fakeMail.getRecipients())
+            .containsOnly(new MailAddress(RCPT_1), new MailAddress(RCPT_2));
+    }
+
+    @Test
+    public void serviceShouldSetToCcAndBccSpecifiedInTheMimeMessage() throws Exception {
+
+        String RCPT_1 = "abc1@apache1.org";
+        String RCPT_2 = "abc2@apache2.org";
+        String RCPT_3 = "abc3@apache3.org";
+
+        FakeMail fakeMail = FakeMail.builder()
+            .recipients(MailAddressFixture.ANY_AT_JAMES)
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .addToRecipient(RCPT_1)
+                .addCcRecipient(RCPT_2)
+                .addBccRecipient(RCPT_3)
+                .build())
+            .build();
+
+        testee.service(fakeMail);
+
+        assertThat(fakeMail.getRecipients())
+            .containsOnly(new MailAddress(RCPT_1), new MailAddress(RCPT_2), new MailAddress(RCPT_3));
+    }
+
+    @Test
+    public void serviceShouldSetEmptyRecipientWhenNoRecipientsInTheMimeMessage() throws Exception {
+
+        FakeMail fakeMail = FakeMail.builder()
+            .recipients(MailAddressFixture.ANY_AT_JAMES)
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .build())
+            .build();
+
+        testee.service(fakeMail);
+
+        assertThat(fakeMail.getRecipients())
+            .isEmpty();
+    }
+
+    @Test
+    public void serviceShouldGoshEmail() throws Exception {
+        FakeMail fakeMail = FakeMail.builder()
+            .recipients(MailAddressFixture.ANY_AT_JAMES)
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .build())
+            .build();
+
+        testee.service(fakeMail);
+
+        assertThat(fakeMail.getState())
+            .isEqualTo(Mail.GHOST);
+    }
+
+    @Test
+    public void serviceShouldResendTheEmail() throws Exception {
+        String RCPT_1 = "abc1@apache1.org";
+        String RCPT_2 = "abc2@apache2.org";
+        String RCPT_3 = "abc3@apache3.org";
+
+        FakeMail fakeMail = FakeMail.builder()
+            .recipients(MailAddressFixture.ANY_AT_JAMES)
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .addToRecipient(RCPT_1)
+                .addCcRecipient(RCPT_2)
+                .addBccRecipient(RCPT_3)
+                .build())
+            .build();
+
+        testee.service(fakeMail);
+
+        assertThat(fakeMail.getRecipients())
+            .containsOnly(new MailAddress(RCPT_1), new MailAddress(RCPT_2), new MailAddress(RCPT_3));
```

Comment:
extract "new MailAddress(RCPT_1)" to constant object then we can reduce line of code